cpp/rss_aggregator.cc

```diff
@@ -176,56 +203,71 @@ unsigned ProcessSection(const std::string &flavour, const IniFile::Section &sect
 const unsigned HARVEST_TIME_WINDOW(60); // days
 
 
-size_t SelectItems(const std::string &flavour, DbConnection * const db_connection,
+struct FeedNameAndURL {
+    std::string name_;
+    std::string url_;
+public:
+    FeedNameAndURL() = default;
+    FeedNameAndURL(const FeedNameAndURL &other) = default;
+    FeedNameAndURL(const std::string &name, const std::string &url)
+        : name_(name), url_(url) { }
+};
+
+
+size_t SelectItems(const std::string &subsystem_type, DbConnection * const db_connection,
                    std::vector<HarvestedRSSItem> * const harvested_items)
 {
-    const auto now(std::time(nullptr));
-    db_connection->queryOrDie("SELECT * FROM rss_aggregator "
-                              "WHERE pub_date >= '" + SqlUtil::TimeTToDatetime(now - HARVEST_TIME_WINDOW * 86400) + "' "
-                              "AND flavour=" + db_connection->escapeAndQuoteString(flavour) + " "
-                              "ORDER BY pub_date DESC");
-    DbResultSet result_set(db_connection->getLastResultSet());
-    while (const DbRow row = result_set.getNextRow())
-        harvested_items->emplace_back(SyndicationFormat::Item(row["item_title"], row["item_description"], row["item_url"], row["item_id"],
-                                                              SqlUtil::DatetimeToTimeT(row["pub_date"])),
-                                                              row["serial_name"], row["feed_url"]);
-    return result_set.size();
+    db_connection->queryOrDie("SELECT id,feed_name,feed_url FROM tuefind_rss_feeds WHERE subsystem_types LIKE '%"
```

Comment:
Why do you use LIKE %...% if we have the exact subsystem type given? (simply use = without %)

Reply:
I don't think that works with sets.